### PR TITLE
Feat: wrap AudioBuffer with JS proxy

### PR DESCRIPTION
**Pull request by b-ma**
### Before

```
RESULTS:
  - # pass: 115
  - # fail: 23
  - # type error issues: 20
 ```
### After 
(w/ https://github.com/orottier/web-audio-api-rs/pull/463) 

```
  RESULTS:
  - # pass: 145
  - # fail: 13
  - # type error issues: 0
```

### CODE REVIEW

File: generator/js/BaseAudioContext.mixin.tmpl.js [hunk 1]
```diff
@@ -40,6 +41,10 @@ ${d.nodes.map(n => `    ${d.name(n)},`).join('\n')}
       }
     }
 
+    createBuffer(numberOfChannels, length, sampleRate) {
```
Comment (orottier) [inline generator/js/BaseAudioContext.mixin.tmpl.js:44]: Good to have this constructor option supported

File: js/lib/cast.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+
+exports.toSanitizedSequence = function toSanitizedSequence(data, targetCtor) {
```
Comment (orottier) [inline js/lib/cast.js:2]: What is this for?

Comment (b-ma) [reply js/lib/cast.js:2]: Finally... for nothing in this PR :) but I was thinking `copyFromChannel` and `copyToChannel` had the same API as for example https://webaudio.github.io/web-audio-api/#dom-baseaudiocontext-createiirfilter which support any kind of floating point value sequence, e.g. Array, Float32Array, Float64Array (numbers is JS are basically only `double`)
Should be useful later I imagine

**Comment by b-ma** [comment]
Just realized there is an issue with `startRendering` and `decodeAudioData` which do not return a wrapped object.
Need to change the wrapping approach to make it more like a mixin rather than use inheritance I guess (e.g. like 
 `AudioParam.js`)

**Comment by orottier** [comment]
Without looking into it, I just want to comment it is possible to both decrease in success and fail.
One newly introduced failure can block many subsequent successes.

I'm using the following trick to digest the results better:

(make sure to run `npm run build` before)

create file `run-wpt.sh` and `chmod +x run-wpt.sh`:
```
#!/bin/bash
echo $1
echo ~~~~~~~~~~~~~~~~~
node ./.scripts/wpt-harness.mjs --filter "$1" 2>&1 | grep -A 3 RESULTS
echo
```

dissect the full suite per directory:
`ls wpt/webaudio/the-audio-api | xargs -I {} ./run-wpt.sh {}`
check the specific file results in a dir:
`ls wpt/webaudio/the-audio-api/the-oscillatornode-interface | xargs -I {} ./run-wpt.sh {}`

**Comment by b-ma** [comment]
Ah thanks!

**Comment by b-ma** [comment]
Ok, finally got it working, it was a bit more complex than expected but paves the way for #84

#### main

```
RESULTS:
  - # pass: 5215
  - # fail: 628
  - # type error issues: 68
```

#### feat/Audiobuffer-proxy

```
RESULTS:
  - # pass: 5262
  - # fail: 603
  - # type error issues: 46
```

**Comment by orottier** [comment]
Cool!

**Pull request merged**